### ISSUE TITLE
support for www.dwiggie.com

### ISSUE THREAD

**Issue by mcepl** [JimmXinu/FanFicFare#903]
www.dwiggie.com is the site dedicated to “Pride and Prejudice” fanfiction community, which also includes fanfiction stories.

Example of a story is 
https://www.dwiggie.com/derby/2019/KathyDeathattheNethe20220403.html

The content is plain HTML, apparently there is no standard for chapters. Most stories seems to be all on one page.

**Comment by kingjon3377** [comment]
Back when I first "packaged" (what became) FanFicFare for Gentoo in [my "overlay"](https://kingjon3377@github.com/kingjon3377/lovelace-overlay), I found a patch containing a DWG ("Derbyshire Writers' Guild", whence "Dwiggie") adapter somewhere (I forget where, and now can't find it), and I've included [that adapter](https://kingjon3377@github.com/kingjon3377/lovelace-overlay/blob/master/app-text/FanFicFare/files/adapter_dwiggiecom.py) in my overlay's packaging of FFF ever since, including adapting it to Python 3 when the time came (which was mercifully a pretty mechanical process---Python is *not* a language I'm competent in!). And I've made good use of it over the years.

However, the DWG (or "dwiggie.com") is included in [the official list of "sites not likely to be supported"](https://github.com/JimmXinu/FanFicFare/wiki/Supportedsites#sites-not-supported) for good reasons, starting with the one listed there:

- There are no chapter lists other than (in *some* stories) a "jump to part X" hyperlink at the top of the first page.
- URLs are not standardized across all stories. (I've routinely had to temporarily change the adapter to get one all the parts of one story in a way that would have broken downloading most other stories.) I'm sure that within any given "era" of the site the URLs are pretty good about adhering to some convention, but the conventions have changed at least twice if not three or four times, and even then there are occasional idiosyncratic exceptions.
- Similarly, the HTML markup within stories (including what has to be parsed to get the essential "metadata," like the title and author name) varies across eras and even occasionally from story to story.
- A story is generally only put into the "archive," as opposed to the "forum," when it is either finished or apparently-abandoned. One of the main points for using FFF rather than a more generic web-to-ebook tool is the ability to add new content to a WIP ebook as the author writes more; that only applies for this site about once a decade.
- The site sees *very* few new stories added to the archive these days. (I have an alert set up on FollowThatPage to check the two "archive index" pages every week or so and email me the diff if there's any change; over the last few years I think there's been no more than five stories added to "Epilogue Abbey" and "Fantasia Gallery" combined, and more stories than that *removed*.)

The problems of non-standardized URLs and HTML could be worked around with heuristics and special cases in the code, but given the other two points above I doubt it's really worth it.
